.github/workflows/firebase-hosting-merge.yml

```diff
@@ -4,17 +4,21 @@
 name: Deploy to Firebase Hosting on merge
 'on':
   push:
-    branches:
-      - main
+    branches: [main, dev]
 jobs:
   build_and_deploy:
     runs-on: ubuntu-latest
+    env:
+      IS_PRODUCTION: ${{ endsWith(github.ref, 'main') }}
+      NODE_OPTIONS: --openssl-legacy-provider
     steps:
       - uses: actions/checkout@v3
       - name: Use Node.js
         uses: actions/setup-node@v3
         with:
-          node-version: '16.x'
+          node-version: '18.x'
+      - name: Set up Firebase CLI
+        run: yarn global add firebase-tools
```

Comment:
referenced how we did it in portal
- node v 16 is not compatible so upgraded to 18